cpp/include/cudf/quantiles.hpp

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.hpp"
+
+namespace cudf {
+
+gdf_column group_quantiles(cudf::table const& input_table,
+                           double quantile,
+                           gdf_context const& context);
```

Comment:
If you're only using `gdf_context` to indicate if the table is already sorted, can you just use a `bool` instead? I'd prefer to not add any new dependencies on `gdf_context`.

Reply:
So now that the PR is taking some shape, I ended up not reinventing the wheel and just using the functions from `groupby_without_aggregation.cu`. I thought I could do away with context but those functions also need it.

Reply:
All the same, I think we should avoid any _external_ dependencies on `gdf_context`. Just accept a `bool` in the `group_quantiles` API and then pass the appropriate `gdf_context` into the necessary APIs.